tests/integ/test_temp_table_cleanup.py

```diff
@@ -139,7 +144,6 @@ def test_reference_count_map_multiple_sessions(db_parameters, session):
         assert session._temp_table_auto_cleaner.ref_count_map[table_name1] == 0
         assert new_session._temp_table_auto_cleaner.ref_count_map[table_name1] == 0
 
-        new_session._temp_table_auto_cleaner.start()
         del df2
         gc.collect()
         time.sleep(WAIT_TIME)
```

Comment:
this wait here seems going to be flaky, one thing you can do is keep track of all asycn_job in the auto_cleaner class, and here you can do a check to see if all job is finished. you can also have a flag to only record this for testing

Reply:
Yes, I will do it in a separate PR (tracked in https://snowflakecomputing.atlassian.net/browse/SNOW-1645523)

Reply:
i see, the whole test is skipped at this moment

Reply:
let's make sure we turn this test on before we start turning the feature on

Reply:
Sure, I still think it's better to do it in next PR, which is much cleaner

Reply:
sue, we can do it in a different pr